src/chemenv/tools/converters.py

```diff
@@ -0,0 +1,387 @@
+        
+        async with aiohttp.ClientSession() as session:
+            try:
+                async with session.get(url, timeout=self.timeout) as response:
+                    if response.status == 200:
+                        return await response.text()
+                    logger.warning(f"CACTUS API failed with status {response.status}")
+                    return None
+            except Exception as e:
+                logger.error(f"CACTUS API error: {e}")
+                return None
+
+    @backoff.on_exception(backoff.expo, (aiohttp.ClientError, asyncio.TimeoutError), max_time=10, logger=logger)
+    async def pubchem(self) -> Optional[str]:
+        """
+        Queries the PubChem REST API to convert chemical name to 
+        isomeric SMILES notation.
+        
+        Returns:
+            str: SMILES notation if successful, None otherwise
+            
+        Raises:
+            aiohttp.ClientError: On API connection errors
+            asyncio.TimeoutError: If request times out
+        """
+        url = f"https://pubchem.ncbi.nlm.nih.gov/rest/pug/compound/name/{self.name}/property/IsomericSMILES/JSON"
+        
+        async with aiohttp.ClientSession() as session:
+            try:
+                async with session.get(url, timeout=self.timeout) as response:
+                    if response.status == 200:
+                        data = await response.json()
+                        return data["PropertyTable"]["Properties"][0]["IsomericSMILES"]
+                    logger.warning(f"PubChem API failed with status {response.status}")
+                    return None
+            except Exception as e:
+                logger.error(f"PubChem API error: {e}")
+                print(e)
+                return None
+
+    @backoff.on_exception(backoff.expo, (aiohttp.ClientError, asyncio.TimeoutError), max_time=10, logger=logger)
+    async def unknown(self) -> Optional[str]:
+        """
+        Queries the Unknown API to convert chemical name to SMILES.
+
+        Returns:
+            str: SMILES notation if successful, None otherwise
+        
+        Raises:
+            aiohttp.ClientError: On API connection errors
+            asyncio.TimeoutError: If request times out
+        """ 
+        url = f"http://46.4.119.202:8082/?name=%7B{self.name}%7D&what=smiles"
+        
+        async with aiohttp.ClientSession() as session:
+            try:
+                async with session.get(url, timeout=self.timeout) as response:
+                    if response.status == 200:
+                        message_content = await response.text()
+                        message_text = message_content.split('Message:')[1].strip()
+                        return message_text
+                    logger.warning(f"Unknown API failed with status {response.status}")
+                    return None
+            except Exception as e:
+                logger.error(f"Unknown API error: {e}")
+                return None
```

Comment:
not sure if we should make this one public

Reply:
What do you think about the current solution? Maybe it can be refined so anyone can use it with a local database or a private endpoint